components/layout_2020/style_ext.rs

```diff
@@ -688,6 +727,19 @@ impl ComputedValuesExt for ComputedValues {
             return true;
         }
 
+        // From <https://www.w3.org/TR/css-will-change/#valdef-will-change-custom-ident>:
+        // > If any non-initial value of a property would cause the element to generate a
+        // > containing block for fixed positioned elements, specifying that property in will-change
+        // > must cause the element to generate a containing block for fixed positioned elements.
+        let will_change_bits = self.clone_will_change().bits;
+        if will_change_bits
+            .intersects(WillChangeBits::PERSPECTIVE | WillChangeBits::FIXPOS_CB_NON_SVG) ||
```

Comment:
`perspective` only applies to transformable elements, so it should be moved under the is_transformable check.